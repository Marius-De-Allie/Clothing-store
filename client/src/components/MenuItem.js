import React, { useCallback } from 'react';
import { useHistory } from 'react-router-dom';
// styles.
import '../styles/menuItem.scss';

const MenuItem = ({ title, imageUrl, size, linkUrl }) => {
    const history = useHistory();

    const onMenuClick = useCallback(() => {
        history.push(`/${linkUrl}`)
    }, [history, linkUrl]);

    return (
        <div 
            className={`${size} menu-item`}
            style={{backgroundImage: `url(${imageUrl})`}}
            onClick={onMenuClick}
        >
            <div 
                className="background-image"
                style={{backgroundImage: `url(${imageUrl})`}}
            >
            </div>
            <div className="content">
                <h1 className="title">{title}</h1>
                <span className="subtitle">SHOP NOW</span>
            </div>
        </div>

    );
};

export default MenuItem;