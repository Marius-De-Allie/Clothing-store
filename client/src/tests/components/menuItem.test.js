import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from '../../components/MenuItem';

it('MenuItem component renders properly', () => {
    const wrapper = shallow(<MenuItem />);
    expect(wrapper).toMatchSnapshot();
});