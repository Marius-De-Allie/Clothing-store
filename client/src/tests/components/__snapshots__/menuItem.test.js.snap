// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component renders properly 1`] = `ShallowWrapper {}`;
